src/AspNet.Security.OAuth.SuperOffice/SuperOfficeAuthenticationOptions.cs

```diff
@@ -0,0 +1,171 @@
+
+        /// <summary>
+        /// Gets or sets the default period of time to cache the SuperOffice public key(s)
+        /// retrieved from the endpoint specified by <see cref="ConfigurationEndpoint"/>.
+        /// </summary>
+        /// <remarks>
+        /// The default public key cache lifetime is 15 minutes.
+        /// </remarks>
+        public TimeSpan PublicKeyCacheLifetime { get; set; } = TimeSpan.FromMinutes(15);
+
+        /// <inheritdoc />
+        public override void Validate()
+        {
+            base.Validate();
+
+            if (_environment == SuperOfficeAuthenticationEnvironment.Production
+                && !AuthorizationEndpoint.StartsWith("https://", System.StringComparison.InvariantCultureIgnoreCase))
+            {
+                throw new SuperOfficeAuthenticationInvalidEnvironmentException("Production environment requires secure endpoints, i.e. begins with 'https://'.");
+            }
+        }
+
+        /// <summary>
+        /// Updates configuration settings.
+        /// </summary>
+        private void UpdateEndpoints()
+        {
+            var env = GetEnvironment();
+
+            AuthorizationEndpoint = string.Format(CultureInfo.InvariantCulture,
+                SuperOfficeAuthenticationConstants.FormatStrings.AuthorizeEndpoint,
+                env);
+
+            TokenEndpoint = string.Format(CultureInfo.InvariantCulture,
+                SuperOfficeAuthenticationConstants.FormatStrings.TokenEndpoint,
+                env);
+
+            ClaimsIssuer = string.Format(CultureInfo.InvariantCulture,
+                SuperOfficeAuthenticationConstants.FormatStrings.ClaimsIssuer,
+                env);
+
+            /* UserInformationEndpoint will include context identifier after authentication
+               in SuperOfficeAuthenticationHandler.CreateTicketAsync */
+
+            UserInformationEndpoint = string.Concat(
+                string.Format(CultureInfo.InvariantCulture, SuperOfficeAuthenticationConstants.FormatStrings.ClaimsIssuer, env),
+                SuperOfficeAuthenticationConstants.FormatStrings.UserInfoEndPoint);
+
+            ConfigurationEndpoint = string.Format(CultureInfo.InvariantCulture,
+                SuperOfficeAuthenticationConstants.FormatStrings.ConfigurationEndpoint,
+                env);
+        }
+
+        /// <summary>
+        /// Used to determine current target environment.
+        /// </summary>
+        /// <returns>Returns the online environment, i.e. development, stage, production.</returns>
+        private string GetEnvironment()
+        {
+            return _environment switch
+            {
+                SuperOfficeAuthenticationEnvironment.Development => "sod",
+                SuperOfficeAuthenticationEnvironment.Stage => "stage",
+                SuperOfficeAuthenticationEnvironment.Production => "online",
+                _ => throw new NotImplementedException() // set in the ctor so never gets here...
```

Comment:
This isn't correct. I could set `Environment` to `(SuperOfficeAuthenticationEnvironment)-1` and this case would be reached.